### PR TITLE
Safari auth session expiration

### PR DESCRIPTION
**Pull request by kirlat**
Improved handling of auth session timeout expiration. Please see #244 for more details.
Due to overall glitchyness of JS objects in Safari after mac is awaken from sleep it is mandatory to do a page refresh to get a consistent UI state.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -36,7 +36,7 @@
     "@babel/core": "^7.6.2",
     "@babel/plugin-transform-runtime": "^7.6.2",
     "@babel/runtime": "^7.6.2",
-    "alpheios-components": "github:alpheios-project/components",
+    "alpheios-components": "github:alpheios-project/components#safari-auth-session-expiration",
```
[the remark this answers is not in the document]
Comment (balmas) [reply package.json]: although really I guess the components fix should be merged into both master and a branch, and then the dependencies fixed in a branch...

[the remark this answers is not in the document]
Comment (kirlat) [reply package.json]: > fix dependency on merge. the dependency should be on v1.4.2 of components, which is the label that will be assigned to the components build with the fix.

Will update dependency once changes in `components` be merged into `master`.

[the remark this answers is not in the document]
Comment (kirlat) [reply package.json]: > although really I guess the components fix should be merged into both master and a branch, and then the dependencies fixed in a branch...

I think I can probably merge this into `3.0.1.82-branch-for-release` first and then `3.0.1.82-branch-for-release` into `master`. Would that be a good approach?

**Comment by kirlat** [comment]
> I wonder if we ought to make similar changes to the background process for the ff/chrome extension too. I suppose it's possible that the timeout could fail to work there too, isn't it?

I think the way state and timeouts are restored is different for Chrome and/or Firefox, comparing with Safari. This behavior might also be platform-dependent, as those browser might be running on Windows, Linux, and macOS at least.

Monica did not report any issues on those platforms so I decided that "if it ain't broke, don't fix it". But I think it is a very good point that we might experience similar issues. However, I think any changes shall be applied only after extensive testing on those platforms. Maybe we can create a separate issue for that and tackle that with a separate PR? I'm just afraid this might be a very big thing to test. What do you think?

File: package.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (balmas) [reply package.json]: yes I think that's fine. thanks

**Comment by balmas** [comment]
> Monica did not report any issues on those platforms so I decided that "if it ain't broke, don't fix it". But I think it is a very good point that we might experience similar issues. However, I think any changes shall be applied only after extensive testing on those platforms. Maybe we can create a separate issue for that and tackle that with a separate PR? I'm just afraid this might be a very big thing to test. What do you think?

Yes that's fine and I think you are probably right to go with the approach of not fixing what isn't reported as broken :-)  Maybe we should just make an issue for it but not worry about fixing it unless it gets reported by a user?

**Comment by kirlat** [comment]
>  Maybe we should just make an issue for it but not worry about fixing it unless it gets reported by a user?

I like this idea! Will do

**Pull request merged**